src/components/date_picker/super_date_picker/super_date_picker.tsx

```diff
@@ -599,3 +621,21 @@ export class EuiSuperDatePicker extends Component<
     );
   }
 }
+
+// Because EuiSuperDatePicker is a class component and not a functional component,
+// we have to use a render prop here in order for us to pass i18n'd strings/objects/etc
+// to all underlying usages of our timeOptions constants. If someday we convert
+// EuiSuperDatePicker to an FC, we can likely get rid of this wrapper.
+export const EuiSuperDatePicker = (props: EuiSuperDatePickerProps) => (
```

Comment:
```suggestion
export const EuiSuperDatePicker: FunctionComponent<EuiSuperDatePickerProps> = (props) => (
```

Not sure that this is any more correct, but the d.ts output has downstream implications

`export const EuiSuperDatePicker: (props: EuiSuperDatePickerProps) => JSX.Element;`
vs
`export const EuiSuperDatePicker: FunctionComponent<EuiSuperDatePickerProps>;`


There are tests in Kibana (`DatePickerWrapper`) that indirectly mock EuiSuperDatePicker and use `typeof EuiSuperDatePicker`. The former does not satisfy the Jest type expectations but the latter does (something about `Constructable`).

Reply:
Great catch, thanks! https://github.com/elastic/eui/pull/5743/commits/83bc9a3f3f927750ce03bb04c913dc4f46adb447